src/services/handler/acknowledgement-handler.js

```diff
@@ -4,21 +4,50 @@ import { setCurrentSpanAttributes } from "../../config/tracing";
 import { ACKNOWLEDGEMENT_TYPES } from "../../constants/acknowledgement-types";
 import { parseAcknowledgementMessage } from "../parser/acknowledgement-parser";
 import { createAcknowledgement } from "../database/create-acknowledgement";
+import { sendDeleteRequestToEhrRepo } from "../ehr-repo/delete-ehr";
+import {
+  getNhsNumberByConversationId,
+  registrationRequestExistsWithMessageId
+} from "../database/registration-request-repository";
 
 export const acknowledgementMessageHandler = async message => {
   const conversationId = await parseConversationId(message);
   const parsedAcknowledgementFields = await parseAcknowledgementMessage(message);
-  const typeCode = parsedAcknowledgementFields.acknowledgementTypeCode;
+  const { acknowledgementTypeCode, messageRef, acknowledgementDetail } = parsedAcknowledgementFields;
+  const isIntegrationAcknowledgement = await registrationRequestExistsWithMessageId(messageRef);
+  const nhsNumber = await getNhsNumberByConversationId(conversationId);
 
   setCurrentSpanAttributes({ conversationId });
 
   await createAcknowledgement(parsedAcknowledgementFields);
 
-  if (ACKNOWLEDGEMENT_TYPES.POSITIVE.includes(typeCode)) {
-    logInfo(`POSITIVE ACKNOWLEDGEMENT RECEIVED`);
-  } else if (ACKNOWLEDGEMENT_TYPES.NEGATIVE.includes(typeCode)) {
-    logInfo(`NEGATIVE ACKNOWLEDGEMENT RECEIVED - DETAIL: ${parsedAcknowledgementFields.acknowledgementDetail}`);
+  if (ACKNOWLEDGEMENT_TYPES.POSITIVE.includes(acknowledgementTypeCode)) {
+    if (isIntegrationAcknowledgement) await handlePositiveIntegrationAcknowledgement(nhsNumber, conversationId);
+    else handlePositiveAcknowledgement();
+  } else if (ACKNOWLEDGEMENT_TYPES.NEGATIVE.includes(acknowledgementTypeCode)) {
+    if (isIntegrationAcknowledgement) handleNegativeIntegrationAcknowledgement();
+    else handleNegativeAcknowledgement(acknowledgementDetail);
   } else {
-    logError(`ACKNOWLEDGEMENT TYPE ${typeCode} IS UNKNOWN.`);
+    logError(`Acknowledgement type ${acknowledgementTypeCode} is unknown.`);
   }
+};
+
+const handlePositiveIntegrationAcknowledgement = async (nhsNumber, conversationId) => {
+  const usefulDetails = `for Conversation ID ${conversationId}, and NHS number ${nhsNumber}.`;
+  logInfo(`Positive integration acknowledgement received ${usefulDetails}`);
+  logInfo(`Sending delete request to ehr out repository ${usefulDetails}`);
+
+  await sendDeleteRequestToEhrRepo(nhsNumber, conversationId);
+};
+
+const handleNegativeIntegrationAcknowledgement = () => {
+  logInfo(`Negative integration acknowledgement received.`);
```

Comment:
James Insley has confirmed that we want negative acknowledgements to also trigger the deletion of the record. We will need to establish at a later date what happens in the event that no integration is ever received.

Reply:
Comment addressed!